song-server/pom.xml

```diff
@@ -240,7 +229,11 @@
             <groupId>javax.xml.bind</groupId>
             <artifactId>jaxb-api</artifactId>
         </dependency>
-
+       <dependency>
+           <groupId>org.glassfish.jaxb</groupId>
+           <artifactId>jaxb-runtime</artifactId>
+           <version>2.3.1</version>
```

Comment:
move this version definition into the `<dependencyManagement>` block of the root pom.xml. If this dependency is only used in song-server, then rip out the dependency from the dependencyManagement section and put it here. Also, use the version variable like `${jaxb.version}`

Reply:
String constants are a valid part of the Java language. Don't be so afraid of them!

If we expect the definition of a constant to ever change, then YES, by all means, we should use a name rather than an inline value.

OBJECT_NAME_DELIMITER might change. NEWLINE might change (based upon different operating systems idea of what a new line should be). 

 But the definition of a something like COMMA should never change, and if it does, it's confusing.

All we accomplish by separating the definition of the constant from the code in this case is forcing the developer to do extra work to double check that COMMA is defined as ",".

Reply:
i find that string constants are harder to read and make it hard to search for via the IDE. by using COMMA, you can find all usages which is very handy when inspecting its usage. Regarding double checking, for something that sensitive, there should be tests in place that ensure it functions as expected.

Reply:
I'm confused. How is searching for "," any harder than searching for COMMA? If anything, it should be easier, since it's two less characters to type. :-) 

There should always be tests. That doesn't mean that the code does what you think it does when you read it.:-) 

But for consistency, I'll make you happy. :-)

Reply:
thank you!